core/src/worker/workflow/machines/workflow_machines.rs

```diff
@@ -629,6 +630,22 @@ impl WorkflowMachines {
                 self.replaying = false;
             }
 
+            if matches!(
+                event.attributes,
+                Some(history_event::Attributes::WorkflowExecutionUpdateRequestedEventAttributes(_)),
+            ) {
+                // The server has sent a durable update requested event: create the message that would have been sent
+                // for a non-durable update request.
+                let msg = IncomingProtocolMessage::try_from(&event)?;
```

Comment:
Would be best to include some additional context (use `.context`) here when bubbling up the error